org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/client/LimitExceededWarnings.java

```diff
@@ -0,0 +1,119 @@
+/*******************************************************************************
+* Copyright (c) 2020 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lemminx.client;
+
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.lemminx.customservice.ActionableNotification;
+import org.eclipse.lemminx.customservice.XMLLanguageClientAPI;
+import org.eclipse.lemminx.settings.SharedSettings;
+import org.eclipse.lsp4j.Command;
+import org.eclipse.lsp4j.MessageParams;
+import org.eclipse.lsp4j.MessageType;
+
+/**
+ * LimitExceededWarnings sends a warning to the
+ * client via a jsonrpc notification
+ */
+public class LimitExceededWarnings {
+
+	/**
+	 * Default instance.
+	 */
+	private static final LimitExceededWarnings DEFAULT_INSTANCE = new LimitExceededWarnings();
+
+	private final Set<String /* file URI */> cache;
+
+	public LimitExceededWarnings() {
+		cache = new HashSet<String>();
+	}
+
+	/**
+	 * Gets the default instance.
+	 *
+	 * @return The default instance
+	 */
+	public static LimitExceededWarnings getDefaultInstance() {
+		return DEFAULT_INSTANCE;
+	}
+
+	/**
+	 * Sends the limit exceeded warning to the client only if the provided
+	 * uri does not exist in the cache
+	 * 
+	 * After sending the warning, the uri is added to the cache
+	 * 
+	 * @param client         the language client
+	 * @param uri            the file uri
+	 * @param resultLimit    the result limit
+	 * @param featureName    the feature name
+	 * @param sharedSettings the shared settings
+	 */
+	public void onResultLimitExceeded(XMLLanguageClientAPI client, String uri, int resultLimit, String featureName,
```

Comment:
replace with 

`public void onResultLimitExceeded(String uri, int resultLimit, String featureName) `

create in this class the enum


```java
public enum LimitFeature {
    
	SYMBOLS("document symbols");
	
	private final String name;
	
	private LimitFeature(String name) {
		this.name = name;
	}
	
	public String getname() {
		return name;
	}
}
```
and add this method

```java
public void onResultLimitExceeded(String uri, LimitFeature feature) {
    int resultLimit = 0;
    switch(feature) {
        case SYMBOLS: 
            resultLimit = sharedSettings.getSymbolSettings().getMaxItemsComputed();
    }
    onResultLimitExceeded(uri, resultLimit, feature.getName());
}
```